### PR TITLE
persist: Remove stale comments

### PR DESCRIPTION
**Pull request by jkosh44**
### Motivation
Update comments.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - There are no user-facing behavior changes.

### CODE REVIEW

File: src/persist-client/src/read.rs [hunk 1]
```diff
@@ -215,13 +214,6 @@ where
     }
 
     /// Politely expires this subscribe, releasing its lease.
-    ///
-    /// There is a best-effort impl in Drop to expire a listen that wasn't
-    /// explicitly expired with this method. When possible, explicit expiry is
-    /// still preferred because the Drop one is best effort and is dependant on
-    /// a tokio [Handle] being available in the TLC at the time of drop (which
-    /// is a bit subtle). Also, explicit expiry allows for control over when it
-    /// happens.
```
Comment (jkosh44) [inline src/persist-client/src/read.rs:224]: As far as I can tell, neither `Listen` not `Subscribe` implement `Drop`.

Comment (bkirwi) [reply src/persist-client/src/read.rs:224]: IIUC the way this is meant to work is that `Listen` has a `ReadHandle`, and the `ReadHandle` implements `Drop`. So IIUC the gist of these is correct but vague... we could probably switch to mentioning the read handle explicitly.

(OTOH, the "These are only here" comments do just seem wrong... lots else going on.

Comment (jkosh44) [reply src/persist-client/src/read.rs]: > IIUC the way this is meant to work is that `Listen` has a `ReadHandle`, and the `ReadHandle` implements `Drop`. So IIUC the gist of these is correct but vague... we could probably switch to mentioning the read handle explicitly.

Ah ok, that makes much more sense. I reverted the comments and just added some clarification to them. Will the `Drop` impl for `ReadHandle` even take care of returning the leased parts of a `Subscribe`?

```
        if let Some(parts) = self.snapshot.take() {
            for part in parts {
                self.return_leased_part(part)
            }
        }
```

**Review by bkirwi — APPROVED** [review]
Rad - thank you for the cleanup!

**Pull request merged**